lib/elastic_record/index/documents.rb

```diff
@@ -164,18 +164,23 @@ def scroll(scroll_id, scroll_keep_alive)
       end
 
       def bulk(options = {})
-        return if current_bulk_batch
-        connection.bulk_actions = []
-
-        yield
-
-        if current_bulk_batch.any?
-          body = current_bulk_batch.map { |action| "#{JSON.generate(action)}\n" }.join
-          results = connection.json_post("/_bulk?#{options.to_query}", body)
-          verify_bulk_results(results)
+        if current_bulk_batch
+          yield
+        else
+          begin
+            connection.bulk_actions = []
+
+            yield
+
+            if current_bulk_batch.any?
+              body = current_bulk_batch.map { |action| "#{JSON.generate(action)}\n" }.join
+              results = connection.json_post("/_bulk?#{options.to_query}", body)
+              verify_bulk_results(results)
+            end
+          ensure
+            connection.bulk_actions = nil
+          end
```

Comment:
I'd like to move the `begin...ensure...end` into its own private method but I don't know what to name it right now.

Reply:
`run_new_bulk_batch`?